app/assets/javascripts/osm.js.erb

```diff
@@ -37,7 +37,8 @@ OSM = {
   SEARCHING:               <%= image_path("searching.gif").to_json %>,
 
   apiUrl: function (object) {
-    var url = "/api/" + OSM.API_VERSION + "/" + object.type + "/" + object.id;
+    var apiType = object.type === "note" ? "notes" : object.type;
```

Comment:
This is kind of ugly... I'm half inclined to say why not just use `notes` as the object type, then you wouldn't need to jump through hoops here to fix things up.

Then again that means internally we are inconsistent about whether the object type is singular or plural, which also offends me ;-)

Reply:
Not using apiUrl() is an option. Then the check for object.type will go to the calling code. "note" as a type is used by iD embedding code. While iD won't display a note, it's being directed to open the note location with links like https://www.openstreetmap.org/edit?note=1000 .